### DISCUSSION TITLE
TNT v0.1.0

### DISCUSSION THREAD

**Discussion by Rishit-dagli** [Rishit-dagli/Transformer-in-Transformer#10] [Announcements]
This is the initial release of TNT TensorFlow and implements Transformers in Transformers as a subclassed TensorFlow model.

## Classes

- [Attention](https://github.com/Rishit-dagli/Transformer-in-Transformer/blob/main/tnt/attention.py): Implements attention as a TensorFlow Keras Layer making some modifications.
- [PreNorm](https://github.com/Rishit-dagli/Transformer-in-Transformer/blob/533c3e45a4a1cfe8c0be03c789fdd2f013186d19/tnt/tnt.py#L20-L28): Normalize the activations of the previous layer for each given example in a batch independently and apply some function to it, implemented as a TensorFlow Keras Layer.
- [FeedForward](https://github.com/Rishit-dagli/Transformer-in-Transformer/blob/533c3e45a4a1cfe8c0be03c789fdd2f013186d19/tnt/tnt.py#L31-L45): Create a FeedForward neural net with two `Dense` layers and GELU activation, implemented as a TensorFlow Keras Layer.
- [TNT](https://github.com/Rishit-dagli/Transformer-in-Transformer/blob/533c3e45a4a1cfe8c0be03c789fdd2f013186d19/tnt/tnt.py#L48-L229): Implements the Transformers in Transformers model using all the other classes, and converts to logits. Implemented as a TensorFlow Keras Model.

<hr /><em>This discussion was created from the release <a href='https://github.com/Rishit-dagli/Transformer-in-Transformer/releases/tag/'>TNT v0.1.0</a>.</em>
